svg.js.d.ts

```diff
@@ -82,58 +105,505 @@ declare module "@svgdotjs/svg.js" {
+        b?: number;
+        c?: number;
+        d?: number;
+        e?: number;
+        f?: number;
+    }
+
+    interface MatrixExtract extends TransformData, MatrixLike { }
+
+    type FlipType = 'both' | 'x' | 'y' | boolean
+    type ArrayXY = [number, number]
+    type CoordinateXY = ArrayXY | { x: number, y: number }
+
+    interface MatrixTransformParam {
+        flip?: FlipType
+        skew?: ArrayXY | number
+        skewX?: number
+        skewY?: number
+        scale?: ArrayXY | number
+        scaleX?: number
+        scaleY?: number
+        shear?: number
+        theta?: number
+        origin?: CoordinateXY
+        around?: CoordinateXY
+        ox?: number
+        originX?: number
+        oy?: number
+        originY?: number
+        position?: CoordinateXY
+        px?: number
+        positionX?: number
+        py?: number
+        positionY?: number
+        translate?: CoordinateXY
+        tx?: number
+        translateX?: number
+        ty?: number
+        translateY?: number
+        relative?: CoordinateXY
+        rx?: number
+        relativeX?: number
+        ry?: number
+        relativeY?: number
+    }
+
+    type MatrixAlias = MatrixLike | TransformData | MatrixTransformParam | number[] | Element | string;
+
+    class Matrix implements MatrixLike {
+        constructor();
+        constructor(source: MatrixAlias);
+        constructor(a: number, b: number, c: number, d: number, e: number, f: number);
+
+        a: number;
+        b: number;
+        c: number;
+        d: number;
+        e: number;
+        f: number;
+
+        clone(): Matrix;
+        transform(o: MatrixLike | MatrixTransformParam): Matrix
+        compose(o: MatrixExtract): Matrix
+        decompose(cx?: number, cy?: number): MatrixExtract
+        multiply(m: MatrixAlias | Matrix): Matrix;
+        lmultiply(m: MatrixAlias | Matrix): Matrix
+        inverse(): Matrix;
+        translate(x?: number, y?: number): Matrix;
+        scale(x: number, y?: number, cx?: number, cy?: number): Matrix;
+        rotate(r: number, cx?: number, cy?: number): Matrix;
+        flip(a: NumberAlias, offset?: number): Matrix;
+        flip(offset?: number): Matrix;
+        shear(a: number, cx?: number, cy?: number): this
+        skew(y?: number, cx?: number, cy?: number): Matrix;
+        skew(x: number, y?: number, cx?: number, cy?: number): Matrix;
+        skewX(x: number, cx?: number, cy?: number): Matrix;
+        skewY(y: number, cx?: number, cy?: number): Matrix;
+        around(cx?: number, cy?: number, matrix?: Matrix): Matrix;
+        equals(m: Matrix): boolean
+        toString(): string;
+        toArray(): number[];
+        valueOf(): MatrixLike;
+        to(a: any): Morphable;
     }
 
-    // boxes.js
-    class Box {
+    // List.js
+    interface List<T> extends Array<T> {
+        // TODO: check whether each is necessary to expose if it useful to external usage.
+        // each(fn: Function, ...args: any[]): void

```

Comment:
Yes, each is totally usable. Its for the case, that you didnt updated the methods in the List prototype and therefore go the way over each. E.g. this will do the same:

```js
list.fill('red')
list.each('fill', 'red')
```

It can also be used to run a custom function over the whole list like map does:

```js
list.each(fn)
```

Ofc you could use map directly in that case... anyway - I will put it in the docs and so it will be public api